client/reader/site-stream/featured.jsx

```diff
@@ -25,12 +25,17 @@ function getPostUrl( post ) {
 	return '/read/blogs/' + post.site_ID + '/posts/' + post.ID;
 }
 
+const FEED_FEATURED_MAX_POST_COUNT = 3;
+
 /* eslint-disable wpcalypso/jsx-classname-namespace */
+// This component is used in the Discover site stream only.
 class FeedFeatured extends React.PureComponent {
 	static displayName = 'FeedFeatured';
 
 	componentDidMount() {
-		this.props.requestPage( { streamKey: this.props.streamKey } );
+		if ( ! this.props.posts || this.props.posts.length < FEED_FEATURED_MAX_POST_COUNT ) {
```

Comment:
Previously, it was possible for the same set of posts to appear twice if the component re-mounted.